doc/releases/changelog-dev.md

```diff
@@ -326,6 +326,10 @@
 
 <h4>Other improvements</h4>
 
+* `default.qubit` now supports adjoint differentiation for any state-based measurement.  Please note that
+  improved perfromance is not guaranteed for non-expectation values.
```

Comment:
Similar comment here as above -- if using VJPs internally, then performance should still be maximally efficient as long as the cost function being differentiated is scalar (so more general than just expectation values)

Reply:
I decided to just leave out this disclaimer from the changelog.